frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/MainActivity.java

```diff
@@ -130,16 +136,47 @@ private static String getMimeType(String url) {
 
     @Override
     public void onConfigurationChanged(Configuration newConfig){
+        int currentNightMode = newConfig.uiMode & Configuration.UI_MODE_NIGHT_MASK;
+        switch (currentNightMode) {
+            case Configuration.UI_MODE_NIGHT_NO:
+                // Night mode is not active, we're using the light theme
+                setTheme(false);
+                break;
+            case Configuration.UI_MODE_NIGHT_YES:
+                // Night mode is active, we're using dark theme
+                setTheme(true);
+                break;
+        }
         super.onConfigurationChanged(newConfig);
     }
 
+    public void setTheme(boolean isDark) {
+        int flags = getWindow().getDecorView().getSystemUiVisibility(); // get current flag
+        if (isDark) {
+            Util.log("Dark theme");
+            getWindow().addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+            getWindow().setStatusBarColor(getResources().getColor(R.color.colorPrimaryDark));
+            getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
+            flags &= ~View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;   // remove LIGHT_STATUS_BAR to flag
+            getWindow().getDecorView().setSystemUiVisibility(flags);
+        } else {
+            Util.log("Light theme");
+            getWindow().addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+            getWindow().setStatusBarColor(getResources().getColor(R.color.colorPrimary));
+            flags |= View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;   // add LIGHT_STATUS_BAR to flag
+            getWindow().getDecorView().setSystemUiVisibility(flags);
+        }
+    }
+
     @SuppressLint("HandlerLeak")
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         Util.log("lifecycle: onCreate");
 
         getSupportActionBar().hide(); // hide title bar with app name.
+        AppCompatDelegate.setDefaultNightMode(MODE_NIGHT_FOLLOW_SYSTEM);
```

Comment:
according to docs this is the default value anyway. was there a reason why you needed to set it?

Reply:
Right, that wasn't needed. Thanks! :)